client-src/src/providers/helper/helper.ts

```diff
@@ -28,4 +29,27 @@ export class HelperProvider {
     return data.sort((a, b) => a - b);
   }
 
+  markInvalidFields(formGroup: FormGroup) {
+    for(let control in formGroup.controls){
+      formGroup.controls[control].markAsTouched();
+      formGroup.controls[control].updateValueAndValidity();
+    }
+    let ionItems = document.getElementsByTagName("ion-item");
+    if(ionItems){
+      for(var i = 0; i < ionItems.length; i++){
+        if (ionItems[i].classList.contains('ng-untouched')) {
+          ionItems[i].classList.remove('ng-untouched');
+          ionItems[i].classList.add('ng-touched');
+        }
+      }
+    }
+    let pinAutoComplete = document.getElementsByTagName("ng2-completer")[0];
+    if(pinAutoComplete){
+      if (pinAutoComplete.classList.contains('ng-untouched')) {
```

Comment:
Should merge the 2 conditions

Reply:
The following conditions?
if(pinAutoComplete){
        if (pinAutoComplete.classList.contains('ng-untouched'))

Reply:
yes

Reply:
Addressed in commit 1ed18e1